File: cpp_e9zMZ/cpp/10-tonghop/125.cpp
#include <math.h>

#include <iostream>
using namespace std;

int main() {
    string str = "abbcccddddeeeeedcba";
    char powerCharacter = str[0];
    int powerTime = 1;
    int result = 0;

    for (int i = 0; i < str.length() - 1; i++) {
        if (powerCharacter == str[i + 1]) {
            powerTime++;
        } else {
            powerCharacter = str[i + 1];
            if (powerTime >= result) result = powerTime;
            powerTime = 1;
        }
    }

    result = max(powerTime, result);
    cout << result;
    return 0;
}

File: python_ToSXs/python/10-tonghop/125.py
str = "cddddeeffffdcba"

powerCharacter = str[0]
powerTime = 1
result = 0

for i in range(len(str) - 1):
    if powerCharacter == str[i + 1]:
        powerTime += 1
    else:
        powerCharacter = str[i + 1]
        if powerTime >= result:
            result = powerTime
        powerTime = 1

result = max(powerTime, result)

print(result)
